django/db/models/lookups.py

```diff
@@ -74,10 +75,21 @@ def set_source_expressions(self, new_exprs):
     def get_prep_lookup(self):
         if hasattr(self.rhs, 'resolve_expression'):
             return self.rhs
-        if self.prepare_rhs and hasattr(self.lhs.output_field, 'get_prep_value'):
-            return self.lhs.output_field.get_prep_value(self.rhs)
+        if not self.prepare_rhs:
+            return self.rhs
+        if hasattr(self.lhs, 'output_field'):
+            if hasattr(self.lhs.output_field, 'get_prep_value'):
+                return self.lhs.output_field.get_prep_value(self.rhs)
+        elif self.rhs_is_direct_value():
+            return Value(self.rhs)
         return self.rhs
 
+    def get_prep_lhs(self):
+        from django.db.models.fields.related_lookups import MultiColSource
+        if hasattr(self.lhs, 'resolve_expression') or isinstance(self.lhs, MultiColSource):
```

Comment:
I wonder if this could be addressed by adding a `MultiColSource` method that returns `self`